confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java

```diff
@@ -327,6 +327,17 @@ public List<TConfigNodeLocation> getOnlineConfigNodes() {
         .collect(Collectors.toList());
   }
 
+  public List<TDataNodeInfo> getOnlineDataNodes(int dataNodeId) {
+    return getNodeManager().getRegisteredDataNodes(dataNodeId).stream()
```

Comment:
Do we need to filter from registeredDataNodes? or could we use the heartbeatCacheMap directly?

Reply:
heartbeatCacheMap only stores the datanodeId and the running status of the datanode，The information of TDataNodeInfo still needs to be obtained from the registeredDataNode